osquery/tables/system/windows/services.cpp

```diff
@@ -42,86 +42,108 @@ const std::map<int, std::string> kServiceType = {
     {0x00000110, "OWN_PROCESS(Interactive)"},
     {0x00000120, "SHARE_PROCESS(Interactive)"}};
 
-bool QuerySvcInfo(const SC_HANDLE& schScManager,
-                  ENUM_SERVICE_STATUS_PROCESS& svc,
-                  Row& r) {
-  DWORD cbBufSize = 0;
+auto closeServiceHandle = [](SC_HANDLE sch) {
+  if (sch != nullptr) {
```

Comment:
I think you get a `nullptr` check for free from the smart pointer's decision to call the dtor?